### PR TITLE
Return logits beamsearch

### PR DESCRIPTION
**Pull request by cmullovisl**
The current beam search code does not accumulate logits across time steps. This pull request implements tracking of the token logit sequence of the current alive beams and returns the logit sequence for the finally chosen hypotheses.

### CODE REVIEW

**Comment by cmullovisl** [comment]
It seems one of the CI jobs failed due to issues unrelated to the pull request ("Could not resolve host: github.com"). Could I ask for a re-run of the failed [build-and-test-cpp-arm64 (macos-15, ruy)](https://github.com/OpenNMT/CTranslate2/actions/runs/22810679290/job/66166967178?pr=2025) job?
